Add the ability to load searchlist variables from a JSON file via the CLI

Swamper78 opened the pull request:
CLI commandline option --json=<file>

My inclination is to use _with open("jsonfile") as f:_ but I thought I'd stick with the existing convention that pickle uses here to explicitly close the file.

Swamper78 committed:
Add loading from JSON file to the CLI

Committer: N Protokowicz <nicklivewire@gmail.com>

Swamper78 committed:
Add loading searchlist variables from a JSON file via the CLI
Fix indent issue.

phdru commented:
Combine the 2 commits and do a force-push. No need to create small related commits.

Please add a test similar for [the test](https://github.com/CheetahTemplate3/cheetah3/blob/a4e33f0f55ed5909437cbbf262fa99de78a7ad2b/Cheetah/Tests/TemplateCmdLineIface.py#L71) for pickle.

Swamper78 commented:
Ok thanks for the feedback.  I'll make the requested changes.

Swamper78 commented:
Hi, I'm still working on this.  I got a little turned around initially failing to note the separation between CheetahWrapper.py (invoked by _cheetah.exe_ from CLI) and TemplateCmdLineIface.py (built into compiled templates invoked by _python mytemplate.py_).

Originally I only added the JSON load function to the cheetah.exe wrapper but now I'm adding it both places, along with the matching unit tests.

While doing this, I've noticed that there does not appear to be any unit test(s) covering the _--pickle_ option in Tests/CheetahWrapper.py as far as I can find, only for the pickle function in the compiled template interface in Tests/TemplateCmdLineIface.py.

phdru commented:
> Hi, I'm still working on this. I got a little turned around initially failing to note the separation between CheetahWrapper.py (invoked by _cheetah.exe_ from CLI) and TemplateCmdLineIface.py (built into compiled templates invoked by _python mytemplate.py_).
> 
> Originally I only added the JSON load function to the cheetah.exe wrapper but now I'm adding it both places, along with the matching unit tests.

Thanks, please go on!

> While doing this, I've noticed that there does not appear to be any unit test(s) covering the _--pickle_ option in Tests/CheetahWrapper.py as far as I can find, only for the pickle function in the compiled template interface in Tests/TemplateCmdLineIface.py.

Not a big deal IMO. If you'd like to work on it — IWBN to fix. If not — no problem.

Swamper78 commented:
Ok, I expect there will be similarities between the JSON and pickle tests in Tests/CheetahWrapper.py so it shouldn't be to much extra for me to get the test for pickle added.

Swamper78 committed:
Add load from JSON file to searchlist + unit tests.  Add missing unit test for pickling in Tests/CheetahWrapper.py

Swamper78 committed:
Add load from JSON file to searchlist + unit tests.  Add missing unit test for pickling in Tests/CheetahWrapper.py

Feature: Add load from JSON file to searchlist + unit tests.  Add test coverage for pickle CLI option.

Swamper78 commented:
Hopefully my code is aligned with the intention of the checkCompile() test, there weren't any other templates in the testing suite with placeholders to use as examples.

phdru commented:
Somehow you removed [one of my commits](https://github.com/CheetahTemplate3/cheetah3/commit/a4e33f0f55ed5909437cbbf262fa99de78a7ad2b) but added the code into [your commit](https://github.com/CheetahTemplate3/cheetah3/pull/39/files#diff-5cf1ddf5c7d5b4abaca8559e0335e34ad421b1b0a9ef491dc186744b3dfe9cc4). Can you please rebase your commit on top of the original commit in branch `master`?

Swamper78 commented:
Sorry, I use Fossil daily so I'm not as versed with git, I might need a little handholding.

It this what you're looking for:
```
git checkout json-cli
git rebase master
```

phdru commented:
> ```
> git checkout json-cli
> git rebase master
> ```

Yes, I think that should work. You can do this in one command:

    git rebase master json-cli

Swamper78 committed:
Merge branch 'json-cli' of https://github.com/Swamper78/cheetah3 into json-cli

phdru commented:
[The result](https://github.com/CheetahTemplate3/cheetah3/pull/39/files) looks good. There is one last problem — there're still too many commits: https://github.com/CheetahTemplate3/cheetah3/pull/39/commits . Can you please combine them into one commit on top of `master`? Without merge…

Swamper78 commented:
Going to close this pull request and open a new one with a single commit from a squashed merge.
